projects/go-lib/src/lib/components/go-button/go-button.component.scss

```diff
@@ -42,15 +42,14 @@
 
 .go-button__icon {
   display: inline-flex;
-  padding-right: .5rem;
 }
 
 .go-button--icon-only {
   padding: calc(.625rem - 1px) .625rem;
+}
 
-  .go-button__icon {
-    padding: 0;
-  }
+.go-button__icon ~ .go-button__text:not(:empty) {
```

Comment:
This seems to have good browser support. The only caveat here is: https://caniuse.com/#feat=mdn-css_selectors_empty_matches_whitespace. Which we shouldn't really have an issue with, but if we do it's linked here for documentation.